com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayersUIBlock.cs

```diff
@@ -17,15 +17,17 @@ public class LayersUIBlock : MaterialUIBlock
     {
         internal class Styles
         {
-            public static readonly string[] headers =
+            public static Vector2 layerIconSize = new Vector2(5, 5);
+
+            public static GUIContent[] layers { get; } =
             {
-                "Main layer",
-                "Layer 1",
-                "Layer 2",
-                "Layer 3",
+                EditorGUIUtility.TrTextContent(" Main layer", icon: Texture2D.whiteTexture),
+                EditorGUIUtility.TrTextContent(" Layer 1", icon: CoreEditorStyles.redTexture),
+                EditorGUIUtility.TrTextContent(" Layer 2", icon: CoreEditorStyles.greenTexture),
+                EditorGUIUtility.TrTextContent(" Layer 3", icon: CoreEditorStyles.blueTexture),
             };
 
-            public static readonly ExpandableBit[] layerExpandableBits =
+            public static ExpandableBit[] layerExpandableBits { get; } =
```

Comment:
I'm interested to know why did you change this read-only field to a property?

Properties are more expensive to access than fields because of the underlying function they generate, furthermore, the data inside the layerExpandableBits can be modified at runtime with a getter like this (which doesn't seem to be useful in this case).

Is there something I'm missing here?

Reply:
I only changed because I didn't like that the field is named layerExpandableBits instead of k_layerExpandable bits. therefore I changed to a property to have a correct naming convention.